### ISSUE TITLE
Distinction between land and sea

### ISSUE THREAD

**Issue by yyylawa** [landlab/landlab#1246]
Sorry for the rudimentary question.

There are components such as LinearDiffuser and FastscapeEroder, but do they affect the ocean and land differently?
If you can make the processes work differently on hillsides and shallow seafloor (up to -15m), I'd be happy.

**Comment by SiccarPoint** [comment]
Neither component has been designed with the seafloor in mind. For the diffuser, using it might make sense if you could justify diffusion as a process on the seafloor, and had an appropriate diffusivity to use. I would be pretty hesitant to use the FastscapeEroder on a sea floor, as it deploys equations only calibrated on terrestrial rivers.

**Comment by yyylawa** [comment]
I have no intention of using FastscapeEroder. However, I would like to do LinearDiffuser on the seafloor for erosion and transport on the shallow seafloor.

I'm afraid that when I use FastscapeEroder or LinearDiffuser for all the grids, it will be perceived as just land, without taking seawater into account.

Is it possible to run the process for a specific elevation (e.g., more than -15m but less than 0m)?
If so, how can this be specified?

**Comment by SiccarPoint** [comment]
Yes, that's what it will do. For something like you're proposing, you would need to modify the grid elevations by hand after each run of the component, to ensure it only affected the elevations you want.

**Comment by yyylawa** [comment]
Oh.
So I need to extract the following grid and run the components, right?
　A: -15m to 0m extracted grid
　B: A grid with 0m or more extracted

I believe that these grid can be extracted by IF statements.

**Comment by SiccarPoint** [comment]
Try that, but be sure to always update the master grad so they can't diverge from it.

**Comment by yyylawa** [comment]
I just have last question about "master grad" .
Does "master grad " mean the slope in the original full grid?
Do you mean that you need to extract the A and B grids again at each step because the 0m position could change?

**Comment by gregtucker** [comment]
Actually there are a couple of ongoing projects that do something like this. The new Sequence model developed by Mike Steckler at LDEO and @mcflugen uses the LinearDiffuser with a spatially variable diffusion coefficient that depends on water depth, and is set to a constant value for locations above sea level. I'm using a similar approach for some exploratory source-to-sink modeling. The component doesn't "know" about land versus sea, but you can write a script that sets the diffusivity coefficient according to location.

**Comment by yyylawa** [comment]
Thanks, SiccarPoint.
In the meantime, I was able to make Linear work only against 0m and above.

Thanks, gregtucker, for the useful information.
I was able to find The new Sequence model. https://sequence.readthedocs.io/en/develop/index.html#

SiccarPoint and gregtucker, Thank you for the kind thoughts.
